File: TestTasks/src/task1/src/runner_node.cpp
#include "ros/ros.h"
#include <geometry_msgs/PoseStamped.h>

using namespace ros;

int main(int argc, char **argv)
{
  init(argc, argv, "runner_node");
  NodeHandle nh;
  Publisher pose_pub = nh.advertise<geometry_msgs::PoseStamped>("pose", 1000);
  Rate loop_rate(10);

  double arg = 0; //main variable for calculation of coordinates
  double speed = 1.5;
  Time prev_time = Time::now();

  while (ok()) {
    //get time delta
    Time cur_time = Time::now();
    double delta_time = (cur_time - prev_time).nsec / pow(10, 9);
    arg += delta_time * speed;

    geometry_msgs::PoseStamped msg;
    msg.header.stamp = cur_time;
    msg.header.frame_id = "map";

    //calculate x,y,z coordinates from arg
    msg.pose.position.x = arg;
    msg.pose.position.y = sin(arg);
    msg.pose.position.z = cos(arg);

    pose_pub.publish(msg);
    prev_time = cur_time;

    spinOnce();
    loop_rate.sleep();
  }

  return 0;
}

File: TestTasks/src/task1/src/runner_node.py
#!/usr/bin/env python

import rospy
import math
from geometry_msgs.msg import PoseStamped


def main():
    rospy.init_node('runner_node', anonymous=True)
    rate = rospy.Rate(10)
    pose_pub = rospy.Publisher('pose', PoseStamped, queue_size=10)

    arg = 0  # main variable for calculation of coordinates
    speed = 1.5
    prev_time = rospy.Time.now()

    while not rospy.is_shutdown():
        # get time delta
        cur_time = rospy.Time.now()
        delta_time = (cur_time - prev_time).nsecs / float(pow(10, 9))
        arg += delta_time * speed

        # calculate x,y,z coordinates from arg
        msg = PoseStamped()
        msg.header.stamp = cur_time
        msg.header.frame_id = "map"
        msg.pose.position.x = arg
        msg.pose.position.y = math.sin(arg)
        msg.pose.position.z = math.cos(arg)

        pose_pub.publish(msg)

        prev_time = cur_time
        rate.sleep()


if __name__ == '__main__':
    main()
